datastore date mismatch

Hi,

I have a test that writes some entries to the datastore and then deletes all the entries that have been written in the past 5 minutes based on a ndb.DateTimeProperty with auto_now_add set to True. If I run a test that writes to the datastore it will set the DateTimeProperty using the UTC time. However, if right after writing to the datastore, I use the datetime.datetime.now() function it will get the date based on my computers timezone (in my case, that's UTC + 2).

Test:
```python
def test_gae(self):

        TaskTrack(id="Test id",
                  snapshot=self.snapshot.key,
                  process="Test",
                  queue="Queue name").put()

        get_query = TaskTrack.query(
            TaskTrack.process == "Test"
        ).fetch()

        print get_query
        print datetime.now()
```

This outputs:

```
[TaskTrack(key=Key('TaskTrack', 'Test id'), enqueued_at=datetime.datetime(2015, 2, 3, 9, 42, 13, 383950), process=u'Test', queue=u'Queue name', snapshot=Key('Snapshot', 20))]

2015-02-03 11:42:13.389502
```

As far as I can tell, using UTC is intended behavior for dev_appserver2.py and the environment should have inherited, but doesn't (please see below a piece of code from dev_appserver2.py). What am I missing?

```python
def main():
  shutdown.install_signal_handlers()
  # The timezone must be set in the devappserver2 process rather than just in
  # the runtime so printed log timestamps are consistent and the taskqueue stub
  # expects the timezone to be UTC. The runtime inherits the environment.
  os.environ['TZ'] = 'UTC'
  if hasattr(time, 'tzset'):
    # time.tzet() should be called on Unix, but doesn't exist on Windows.
    time.tzset()
  options = PARSER.parse_args()
  dev_server = DevelopmentServer()
  try:
    dev_server.start(options)
    shutdown.wait_until_shutdown()
  finally:
    dev_server.stop()
```

This is my model:

```python
class TaskTrack(BaseModel):
    enqueued_at = ndb.DateTimeProperty(auto_now_add=True)
    snapshot = ndb.KeyProperty(kind=Snapshot)
    process = ndb.StringProperty()
    queue = ndb.StringProperty(required=True)
```

I did a quick test on my system where I do not have the `TZ` environment variable set but my local time is America/New_York.

```python
    def test_datetime(self):
        from google.appengine.ext import ndb
        from datetime import datetime
        class TZTest(ndb.Model):
            the_time = ndb.DateTimeProperty(auto_now_add=True)

        me = TZTest()
        me.put()

        self.assertLess(me.the_time, datetime.now())
```

Here are the results:
```
======================================================================
FAIL: test_datetime (tests.test_timezone)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/Josh/Developer/nosegae_examples/tests/test_timezone.py", line 250, in test_datetime
    self.assertLess(me.the_time, datetime.now())
AssertionError: datetime.datetime(2015, 2, 3, 15, 19, 55, 163368) not less than datetime.datetime(2015, 2, 3, 10, 19, 55, 165160)
```

In my own testing, I use [fixture](https://github.com/fixture-py/fixture) for setting up test data instead of adding entities directly. That way the data is removed for me once my tests are complete.

That being said, the `TZ` issue needs to be resolved because any date or time objects created by the appengine framework would be UTC while your own objects would be whatever local time your system was set to.

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,10 @@
+0.4.1
+
+- Fix issue with timezone mismatch discovered by viorelsfetea by importing
+  google.appengine.tools.os_compat which explicitly sets the timezone to UTC
+  so that datetime and time objects created in tests match the ones created
+  in the appengine runtime.
+
 0.4.0
 
 - Huge rewrite to support dev_appserver2

--- a/LICENSE.txt
+++ b/LICENSE.txt
@@ -2,10 +2,22 @@ Copyright (c) 2015, Joshua Johnston
 
 All rights reserved.
 
-Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
 
-1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+1. Redistributions of source code must retain the above copyright notice, this list
+   of conditions and the following disclaimer.
 
-2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+2. Redistributions in binary form must reproduce the above copyright notice, this list
+   of conditions and the following disclaimer in the documentation and/or other materials
+   provided with the distribution.
 
-THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

--- a/nosegae.py
+++ b/nosegae.py
@@ -5,6 +5,22 @@ import tempfile
 from nose.plugins.base import Plugin
 from nose.case import FunctionTestCase
 
+# Solution from
+# http://stackoverflow.com/questions/17583443/what-is-the-correct-way-to-share-package-version-with-setup-py-and-the-package
+from pkg_resources import get_distribution, DistributionNotFound
+
+try:
+    _dist = get_distribution('nosegae')
+    # Normalize case for Windows systems
+    dist_loc = os.path.normcase(_dist.location)
+    here = os.path.normcase(__file__)
+    if not here.startswith(os.path.join(dist_loc, 'nosegae')):
+        # not installed, but there is another version that *is*
+        raise DistributionNotFound
+except DistributionNotFound:
+    __version__ = 'DEVELOPMENT'
+else:
+    __version__ = _dist.version
 
 logger = logging.getLogger(__name__)
 
@@ -61,6 +77,11 @@ class NoseGAE(Plugin):
 
         import dev_appserver
         dev_appserver.fix_sys_path()  # add paths to libs specified in app.yaml, etc
+
+        # This file handles some OS compat settings. Most notably the `TZ` stuff
+        # to resolve https://github.com/Trii/NoseGAE/issues/14.
+        # It may need to be removed in the future if Google changes the functionality
+        import google.appengine.tools.os_compat
 
         from google.appengine.tools.devappserver2 import application_configuration
 

--- a/setup.py
+++ b/setup.py
@@ -1,16 +1,8 @@
 from setuptools import setup
 
-DESCRIPTION = """
-NoseGAE: nose plugin for Google App Engine testing
-"""
-LONG_DESCRIPTION = """
-Basic usage:
-    
-    $ cd your/app
-    $ nosetests --with-gae
+DESCRIPTION = "NoseGAE: nose plugin for Google App Engine testing"
 
-"""
-VERSION = '0.4.0'
+VERSION = '0.4.1'
 
 setup(
     name='NoseGAE',
@@ -20,21 +12,22 @@ setup(
     maintainer="Josh Johnston",
     maintainer_email="johnston.joshua+nosegae@gmail.com",
     description=DESCRIPTION,
-    long_description=LONG_DESCRIPTION,
+    setup_requires=['setuptools-markdown'],
+    long_description_markdown_filename='README.md',
     classifiers=[
         "Development Status :: 4 - Beta",
         "Environment :: Console",
         "Intended Audience :: Developers",
         "License :: OSI Approved :: BSD License",
         "Topic :: Software Development :: Testing"
-        ],
+    ],
     url='https://github.com/Trii/NoseGAE',
     download_url='https://github.com/Trii/NoseGAE/tarball/' + VERSION,
     license='BSD License',
     entry_points={
         'nose.plugins.0.10': ['nosegae = nosegae:NoseGAE']
-        },
+    },
     py_modules=['nosegae'],
     install_requires=['nose>=0.10.1'],
     zip_safe=False
-    )
+)

